apps/google-analytics-4/lambda/src/middlewares/service-account-key-provider.ts

```diff
@@ -0,0 +1,84 @@
+import { RequestHandler } from 'express';
+import { ServiceAccountKeyFile, ServiceAccountKeyId } from '../types';
+
+export class MissingServiceAccountKeyHeader extends Error {}
+export class InvalidServiceAccountKey extends Error {}
+
+export const serviceAccountKeyProvider: RequestHandler = (req, _res, next) => {
+  const serviceAccountKeyIdHeaderValue = req.headers['x-contentful-serviceaccountkeyid'];
+  if (!serviceAccountKeyIdHeaderValue || typeof serviceAccountKeyIdHeaderValue !== 'string') {
+    throw new MissingServiceAccountKeyHeader(
+      "missing or incorrectly formatted header 'x-contentful-serviceaccountkeyid'"
+    );
+  }
+  const serviceAccountKeyId = decodeServiceAccountHeader(serviceAccountKeyIdHeaderValue);
+  assertServiceAccountKeyId(serviceAccountKeyId);
+  req.serviceAccountKeyId = serviceAccountKeyId;
+
+  // Pulling the account key from the header will be removed later and replaced with fetching
+  // the code directly from secret storage!
+  const serviceAccountKeyHeaderValue = req.headers['x-contentful-serviceaccountkey'];
+  if (!serviceAccountKeyHeaderValue || typeof serviceAccountKeyHeaderValue !== 'string') {
```

Comment:
I think this typeof check is superfluous unless you were dealing with express framework typecasting header values?

Reply:
yeah, that's the issue. express says this could be a string _or_ an array of strings. e.g if your request is -h "foo=bar" -h "foo=baz" then your header `request.headers['foo']` is going to be `['bar', 'baz']`.

Reply:
yep, k